go.mod

```diff
@@ -1,63 +1,219 @@
 module github.com/viamrobotics/agent
 
-go 1.23.8
+go 1.24
+
+toolchain go1.24.9
+
+tool github.com/golangci/golangci-lint/cmd/golangci-lint
 
 require (
 	github.com/Masterminds/semver/v3 v3.3.1
 	github.com/gabriel-vasile/mimetype v1.4.9
 	github.com/godbus/dbus/v5 v5.1.1-0.20241109141230-b9236d654833
 	github.com/google/uuid v1.6.0
+	github.com/hashicorp/go-getter v1.8.2
 	github.com/jessevdk/go-flags v1.6.1
 	github.com/nightlyone/lockfile v1.0.0
 	github.com/pkg/errors v0.9.1
 	github.com/samber/mo v1.16.0
 	github.com/schollz/progressbar/v3 v3.18.0
 	github.com/sergeymakinen/go-systemdconf/v2 v2.0.2
 	github.com/tidwall/jsonc v0.3.2
-	github.com/ulikunitz/xz v0.5.14
+	github.com/ulikunitz/xz v0.5.15
 	github.com/viamrobotics/gonetworkmanager/v2 v2.2.3
 	go.uber.org/zap v1.27.0
 	go.viam.com/api v0.1.443
 	go.viam.com/rdk v0.73.0
 	go.viam.com/test v1.2.4
 	go.viam.com/utils v0.1.142
-	golang.org/x/sys v0.32.0
+	golang.org/x/sys v0.35.0
 	google.golang.org/grpc v1.72.0
 	google.golang.org/protobuf v1.36.6
 	tinygo.org/x/bluetooth v0.11.0
 )
 
 require (
+	4d63.com/gocheckcompilerdirectives v1.2.1 // indirect
```

Comment:
Josh brought up offline, but... why are there _so_ many transitive dependencies added here? Does the size of our generated agent binary go up because of all of these dependencies? I see [an old issue here](https://github.com/hashicorp/go-getter/issues/193) that seems to reference that.

Reply:
okay this PR is Bad :tm: if the binary size increased significantly; I'm going to check and get back to you. the point of this project is to improve downloads on bad connections, so bigger binary would not be great

(and thank you for looking into this)

fwiw I wrote an in-house version of this + then closed the PR, so it will not be hard to fall back if we need to https://github.com/viamrobotics/rdk/pull/5375

Reply:
Thanks for looking into a fork! What's the final size with our fork again?